examples/client.rs

```diff
@@ -31,76 +33,85 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
         .with_span_events(tracing_subscriber::fmt::format::FmtSpan::FULL)
         .with_writer(std::io::stderr)
+        .with_max_level(tracing::Level::INFO)
         .init();
 
     let opt = Opt::from_args();
 
-    let dest = opt.uri.parse::<http::Uri>()?;
+    // DNS lookup
 
-    if dest.scheme() != Some(&http::uri::Scheme::HTTPS) {
-        Err("destination scheme must be 'https'")?;
+    let uri = opt.uri.parse::<http::Uri>()?;
+
+    if uri.scheme() != Some(&http::uri::Scheme::HTTPS) {
+        Err("uri scheme must be 'https'")?;
     }
 
-    let auth = dest
-        .authority()
-        .ok_or("destination must have a host")?
-        .clone();
+    let auth = uri.authority().ok_or("uri must have a host")?.clone();
 
     let port = auth.port_u16().unwrap_or(443);
 
-    // dns me!
     let addr = tokio::net::lookup_host((auth.host(), port))
         .await?
         .next()
         .ok_or("dns found no addresses")?;
 
-    eprintln!("DNS Lookup for {:?}: {:?}", dest, addr);
+    info!("DNS lookup for {:?}: {:?}", uri, addr);
 
-    // quinn setup
-    let tls_config_builder = rustls::ClientConfig::builder()
-        .with_safe_default_cipher_suites()
-        .with_safe_default_kx_groups()
-        .with_protocol_versions(&[&rustls::version::TLS13])?;
-    let mut tls_config = if !opt.insecure {
-        let mut roots = rustls::RootCertStore::empty();
-        match rustls_native_certs::load_native_certs() {
-            Ok(certs) => {
-                for cert in certs {
-                    if let Err(e) = roots.add(&rustls::Certificate(cert.0)) {
-                        eprintln!("failed to parse trust anchor: {}", e);
-                    }
+    // create quinn client endpoint
+
+    // load CA certificates stored in the system
+    let mut roots = rustls::RootCertStore::empty();
+    match rustls_native_certs::load_native_certs() {
+        Ok(certs) => {
+            for cert in certs {
+                if let Err(e) = roots.add(&rustls::Certificate(cert.0)) {
+                    error!("failed to parse trust anchor: {}", e);
                 }
             }
-            Err(e) => {
-                eprintln!("couldn't load any default trust roots: {}", e);
-            }
-        };
-        tls_config_builder
-            .with_root_certificates(roots)
-            .with_no_client_auth()
-    } else {
-        tls_config_builder
-            .with_custom_certificate_verifier(Arc::new(YesVerifier))
-            .with_no_client_auth()
+        }
+        Err(e) => {
+            error!("couldn't load any default trust roots: {}", e);
+        }
     };
+
+    // load certificate of CA who issues the server certificate
+    if let Err(e) = roots.add(&rustls::Certificate(std::fs::read(opt.ca)?)) {
```

Comment:
Good job the examples are simpler now.

Do we need a warning here that we do that for the example, but to think twice before doing that in production code?
Also iirc this may enable Man-in-the-middle attack when using the example client.

Reply:
done